source/_patterns/01-elements/input/input.scss

```diff
@@ -1,50 +1,145 @@
 @charset "utf-8";
 
 @use "@db-ui/base/build/scss/variables" as *;
+@use "@db-ui/base/build/scss/variables.global" as *;
 @use "@db-ui/base/build/scss/helpers/functions" as *;
-@import "input.variables";
-@import "../form-elements";
+@use "@db-ui/base/build/scss/icon/icons.helpers" as *;
+
+$icons-path: "../../../icons/" !default;
+
+/* stylelint-disable */
+
+@mixin label-focus-animation($translationX: var(--elm-input-padding-left)) {
+	$transformScale: -1.25;
+
+	transition: opacity 0.08s, transform 0.15s;
+	opacity: var(--db-input-focus-opacity);
+
+	transform: translate($translationX, calc($db-sizing-md * $transformScale))
+		scale(var(--db-input-label-scale));
+}
 
 .elm-input {
-	@extend %form-element;
+	position: relative;
 
-	color: $input---color;
+	--elm-input-padding-left: #{$db-spacing-fixed-sm};
+	--elm-input-padding-right: #{$db-spacing-fixed-sm};
 
-	font-size: to-rem($pxValue: 16);
-	height: to-rem($pxValue: 44);
-	line-height: to-rem($pxValue: 24);
-	max-width: 100%;
+	&[data-icon-before] {
+		--elm-input-padding-left: #{calc(2 * $db-spacing-fixed-sm + 0.425rem)};
+
+		input {
+			padding-left: var(--elm-input-padding-left);
+		}
+	}
+
+	&[data-icon-after] {
+		--elm-input-padding-right: #{calc(
+				2 * #{$db-spacing-fixed-sm} + 0.425rem
+			)};
+
+		input {
+			padding-right: var(--elm-input-padding-right);
+		}
+
+		.elm-state-icon::after {
+			margin-right: calc(0.75rem + $db-spacing-fixed-xs);
+		}
+	}
+
+	&[data-variant][data-icon-after] {
+		--elm-input-padding-right: #{calc(
+				2 * #{$db-spacing-fixed-sm} + 0.425rem + 0.75rem + #{$db-spacing-fixed-xs}
+			)};
+
+		input {
+			padding-right: var(--elm-input-padding-right);
+		}
+	}
+
+	&::before,
+	&::after {
+		@extend %db-body-sm;
+		--icon-font-family: var(--db-base-icon-font-family);
+		--icon-font-size: var(--db-base-icon-font-size);
+
+		height: var(--db-sizing-md);
+		display: flex;
+		align-items: center;
+		position: absolute;
+		top: 0;
+		padding: 0 $db-spacing-fixed-xs;
+	}
+
+	&::after {
+		right: 0;
+	}
+}
 
-	padding: $db-spacing-fixed-regular-xs $db-spacing-fixed-regular-m;
+.elm-input input {
+	@extend %db-bg-neutral-transparent-semi;
 
+	border: none;
```

Comment:
If we don't need a border, we should use `border-color: transparent` instead for Windows High Contrast mode reasons: https://piccalil.li/quick-tip/use-transparent-borders-and-outlines-to-assist-with-high-contrast-mode/

Reply:
changed in https://github.com/db-ui/core/pull/198/commits/9291c0e751599052a5c87dc858f92b0e719c1eaa